packages/core/source/units.ts

```diff
@@ -295,7 +295,7 @@ export function areUnitConvertible(a: Unit | undefined, b: Unit | undefined) {
 		b.numerators,
 		b.denominators,
 	].map(countByUnitClass)
-	const uniq = <T>(arr: Array<T>): Array<T> => [...new Set(arr)]
+	const uniq = <T>(arr: Array<T>): Array<T> => Array.from(new Set(arr))
```

Comment:
Woah that was... unexpected. No idea what's going on here.
This is working perfectly fine in dev mode, but there must be a wrong transformation of the code in prod mode (by babel?).

In prod: `uniq([1, 1]) === [new Set([ 1 ])]` 🥴

Maybe this could be fixed by upgrading the docusaurus/webpack/babel/whatever packages, but I can't test in #290 has the studio is currently broken there.

Reply:
wow

Reply:
#290 bug cause of something similar: 
```js
const dependencies = [
		...(engine.context.referencesMaps.referencesIn.get(dottedName) ?? []),
	]
```
is convert into  
```js
[].concat(engine.context.referencesMaps.referencesIn.get(dottedName) ?? [])
```  
in prod...

Reply:
I don't think `tsup` is involved in the docusaurus app? (AFAIK it's only used to bundle the packages, which is working fine)

Reply:
It's not in docusaurus, it's in react-ui package, in my pr i add tsup in react-ui.
And it's only in prod too, so it's the builder I think, so tsup no ?

Reply:
My bad, i try in [babel studio](https://babeljs.io/repl/#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=usage&corejs=3.21&spec=true&loose=true&code_lz=MYewdgzgLgBAngmBeGBtAdJsBTA7jAZWygAoBKAXSA&debug=false&forceAllTransforms=true&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2%2Ctypescript&prettier=false&targets=Node-10.13&version=7.20.6&externalPlugins=&assumptions=%7B%7D) and it seems to be the "force all transforms" option

Reply:
I think the babel config from docusaurus cause the issue with the `loose: true` option [here](https://github.com/facebook/docusaurus/blob/89b0fff128d2272714f786b4286d49a410268928/packages/docusaurus/src/babel/preset.ts#L33) 
> In loose mode, all iterables are assumed to be arrays.

source: https://github.com/babel/babel/issues/7958#issuecomment-389916364

https://github.com/facebook/docusaurus/issues/7606

Reply:
`loose: true` by default  😮 (but only in prod so its undetectable in dev)
https://www.youtube.com/watch?v=aXOChLn5ZdQ

Reply:
Yeah ^^

Reply:
I post a comment here to know a way to fix that : https://github.com/facebook/docusaurus/issues/7606#issuecomment-1330452598
for now the workaround is like you do : `Array.from(new Set())`